src/monodroid/jni/monodroid-glue.c

```diff
@@ -535,6 +536,28 @@ get_xamarin_android_msbuild_path (void)
 	return msbuild_folder_path;
 }
 
+static char *libmonoandroid_directory_path = NULL;
+
+// Returns the directory in which this library was loaded from
+static char*
+get_libmonoandroid_directory_path ()
+{
+	wchar_t module_path[MAX_PATH];
+	char *converted_path = NULL;
+	HMODULE module = NULL;
+
+	if (libmonoandroid_directory_path != NULL)
+		return libmonoandroid_directory_path;
+
+	if (!GetModuleHandleEx (GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
+	                        (LPCSTR)&libmonoandroid_directory_path, &module))
+		return NULL;
+	GetModuleFileNameW (module, module_path, sizeof (module_path));
+	converted_path = utf16_to_utf8 (module_path);
+	libmonoandroid_directory_path = dirname (converted_path);
```

Comment:
So I guess since we both dislike `dirname` and I don't feel good with `strchr` we can go for either [PathRemoveFileSpec](https://msdn.microsoft.com/en-us/library/windows/desktop/bb773748(v=vs.85).aspx) or [PathCchRemoveFileSpec](https://msdn.microsoft.com/en-us/library/windows/desktop/hh707092(v=vs.85).aspx). The first is deprecated in favor of the second but that second one is only on Win 8+, do we still have people using our stuff in Windows 7?